### PR TITLE
Thread optional yield context through  get_bucket_info call path

### PR DESCRIPTION
**Pull request by alimaredia**
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by alimaredia** [comment]
Here is a teuthology run with this branch: http://pulpito.ceph.com/amaredia-2019-05-03_03:24:53-rgw-wip-rgw-thread-yield-context-get-bucket-info-distro-basic-smithi/

Everything looks fine to me.

File: src/rgw/rgw_op.cc [hunk 1]
```diff
@@ -3149,7 +3149,7 @@ void RGWCreateBucket::execute()
       map<string, bufferlist> battrs;
 
       op_ret = store->get_bucket_info(*s->sysobj_ctx, s->bucket_tenant, s->bucket_name,
-                                      binfo, nullptr, &battrs);
+                                      binfo, nullptr, null_yield, &battrs);
```
Comment (cbodley) [inline src/rgw/rgw_op.cc]: a few of these null_yields in rgw_op.cc could be s->yield

**Comment by cbodley** [comment]
jenkins test make check

**Comment by alimaredia** [comment]
Here is a teuthology run: http://pulpito.ceph.com/amaredia-2019-05-03_03:24:53-rgw-wip-rgw-thread-yield-context-get-bucket-info-distro-basic-smithi/

**Comment by alimaredia** [comment]
jenkins test make check

**Comment by cbodley** [comment]
the make check failure is tracked in https://tracker.ceph.com/issues/39995

**Comment by cbodley** [comment]
jenkins test make check

**Pull request merged**